### PR TITLE
Added first_time_commenters function

### LINKED ISSUE #3509: Method to check if user has commented for first-time or not
(This issue is a part of issue #3331 )

There is a method in https://github.com/publiclab/plots2/blob/master/app/models/user.rb#L343 to check if a user is a first-time poster or not. But, this method checks only for nodes(question, research note, etc). 

You have to make a method by the name `first_time_commenter` and write it's definition as to check if user has commented before or not. Note - You have to check this based on the number of published comments i.e., comment with `status = 1`. Place this method beneath the first_time_poster method only. 

Feel free to ask, if you have any queries. Thanks :smiley: !

### PR DESCRIPTION
**Pull request by saloni-garg**
Fixes #3509 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by saloni-garg** [comment]
@gauravano, Please have a look :)

**Comment by gauravano** [comment]
Hi @saloni-garg, I think the above code requires a little change, how about you run the code in rails console and see the output. Thanks!

**Comment by saloni-garg** [comment]
Okay, @gauravano :)

File: app/models/user.rb [hunk 1]
```diff
@@ -343,6 +343,10 @@ def photo_path(size = :medium)
   def first_time_poster
     notes.where(status: 1).count == 0
   end
+  
+  def first_time_commenter
+    comments.where(status: 1).count == 0
```
Comment (gauravano) [inline app/models/user.rb]:
```suggestion
    Comment.where(status: 1).count == 0
```

**Comment by jywarren** [comment]
Awesome!!!! Approving and merging. Great work!

File: app/models/user.rb [hunk 2]
```diff
@@ -343,6 +343,10 @@ def photo_path(size = :medium)
   def first_time_poster
     notes.where(status: 1).count == 0
   end
+  
+  def first_time_commenter
+    Comment.where(status: 1).count == 0
```
Comment (gauravano) [inline app/models/user.rb]: This line is wrong and line is updated in #4098 to `Comment.where(status: 1, uid: uid).count == 0`.

**Pull request merged**